### PR TITLE
APEXMALHAR-2057 :  CSVParser fails continuously when invalid record is passed- Issue Fixed

### PR DESCRIPTION
**Pull request by shubham-pathak22**

### CODE REVIEW

**Comment by bhupeshchawda** [comment]
Can you please add a unit test?

**Comment by shubham-pathak22** [comment]
Added unit tests.

**Comment by bhupeshchawda** [comment]
Can you also remove the checkstyle related changes from this pull request. Let's restrict this to only changes related to this bug fix.

**Comment by shubham-pathak22** [comment]
Removed  check style related changes

**Comment by shubham-pathak22** [comment]
@bhupeshchawda Have incorporated the comments.

File: contrib/src/test/java/com/datatorrent/contrib/parser/CsvPOJOParserTest.java [hunk 1]
```diff
@@ -319,6 +331,19 @@ public void TestParserMoreFields()
...
+  public void TestParserMoreFieldsOnlyPOJOPortConnected()
+  {
+    parser.parsedOutput.setSink(null);
+    parser.beginWindow(0);
+    parser.in.process(
+        "1234,98233,adxyz,0.2,2015-03-08 03:37:12,11/12/2012,12,y,OPTIMIZATION,CAMP_AD,Y,yes,ExtraField".getBytes());
+    parser.endWindow();
+    Assert.assertEquals(0, objectPort.collectedTuples.size());
+    Assert.assertEquals(0, pojoPort.collectedTuples.size());
+    Assert.assertEquals(1, error.collectedTuples.size());
```
Comment (gauravgopi123) [inline contrib/src/test/java/com/datatorrent/contrib/parser/CsvPOJOParserTest.java:374]: Can you add small addition to make sure that error port has same string tuple that was passed on input port?

File: contrib/src/test/java/com/datatorrent/contrib/parser/CsvPOJOParserTest.java [hunk 2]
```diff
@@ -292,30 +313,68 @@ public void TestParserHeaderAsInput()
...
+  {
+    String input = "1234,98233,adxyz,0.2,2015-03-08 03:37:12,11/12/2012,12,y,OPTIMIZATION,CAMP_AD,Y,yes,ExtraField";
+    parser.parsedOutput.setSink(null);
+    parser.beginWindow(0);
+    parser.in.process(input.getBytes());
     parser.endWindow();
     Assert.assertEquals(0, objectPort.collectedTuples.size());
     Assert.assertEquals(0, pojoPort.collectedTuples.size());
     Assert.assertEquals(1, error.collectedTuples.size());
+    Assert.assertEquals(1, error.collectedTuples.size());
```
Comment (sandeepdeshmukh) [inline contrib/src/test/java/com/datatorrent/contrib/parser/CsvPOJOParserTest.java]: Looks like  duplicate line, same as 374. Could you please remove that.

**Pull request merged**